packages/graphql/src/translate/create-update-and-params.ts

```diff
@@ -469,12 +478,31 @@ function createUpdateAndParams({
                 } else {
                     res.strs.push(`SET ${varName}.${dbFieldName} = point($${param})`);
                 }
+            } else if (isMathField) {
+                const mathProperty = mathFieldMatch[1];
+                const mathOperatorName = mathFieldMatch[2];
+                const mathSymbol = mathOperatorToSymbol(mathOperatorName);
+                if (mathSymbol === "/" && value === 0) {
+                    throw new Error('Division by zero is not supported');
+                }
+                if (updateInput[mathProperty]) {
+                    throw new Error(`Ambigous property: ${mathProperty}`);
+                }
+
+                const fieldType = settableField.typeMeta.name;
+                const bitSize = fieldType === "Int" ? 32 : 64;
+                // Avoid overflows, for 64 bit overflows, a long overflow is raised anyway by Neo4j
+                res.strs.push(`WITH ${varName} CALL apoc.util.validate(${varName}.${mathProperty} ${mathSymbol} $${param} > 2^${bitSize-1}-1, 'Value returned from operator %s is larger than %s bit', ["${mathOperatorName}", "${bitSize}"])`);
+                const cyperType = fieldType === "Int" || fieldType === "BigInt" ? "INTEGER" : "FLOAT";
+                // Avoid type coercion
+                res.strs.push(`WITH ${varName} CALL apoc.util.validate(apoc.meta.type(${varName}.${mathProperty} ${mathSymbol} $${param}) <> "${cyperType}", 'Value returned from operator %s does not match: %s', ["${mathOperatorName}", "${fieldType}"])`);
+                res.strs.push(`SET ${varName}.${mathProperty} = ${varName}.${mathProperty} ${mathSymbol} $${param}`);
```

Comment:
Consider moving this whole block into a separate function to reduce the size of the createUpdateAndParams function